### PR TITLE
Generic tool improvements

### PR DESCRIPTION
**Pull request by dymurray**
* Added `apb setup`
  * Creates user `apb-developer` with `cluster-admin` rights
* Added sanity checks for --broker suffix so that `<route>/ansible-service-broker` isn't required and just `<route>` will work.
*  make `apb push -o` the default `apb push` and old `apb push` functionality is `apb push --push-to-broker`.

### CODE REVIEW

**Comment by dymurray** [comment]
https://bugzilla.redhat.com/show_bug.cgi?id=1523252

File: src/apb/engine.py
Comment (dymurray) [inline src/apb/engine.py:1106]: This is causing errors without the last slash. It was doing `<route>`/v2/apb<apb_id>`.

File: src/apb/engine.py
Comment (dymurray) [inline src/apb/engine.py:1053]: flag is `--push-to-broker`

**Comment by rthallisey** [comment]
> Deprecate apb push and make apb push -o the default
>
We are not doing this any more right?

File: src/apb/cli.py
Comment (rthallisey) [inline src/apb/cli.py]: I dropped ```/spec```, it's now ```/v2/apb```.

**Comment by dymurray** [comment]
@rthallisey you're right. I updated to reflect the changes.

File: src/apb/engine.py
Comment (rthallisey) [inline src/apb/engine.py]: Should this print still be here?

**Comment by rthallisey** [comment]
We'll definitely need to get this in before I can merge my PR that adds classes

**Comment by dymurray** [comment]
@rthallisey thanks I had intended on printing the current logged in user but ran into issues didn't want to fight that. I plan on adding more to `apb setup` as we find the need for it.

**Review by rthallisey — APPROVED** [review]

**Comment by rthallisey** [comment]
@dymurray I'm good with merging this.

Also, if you can hold off on any apb changes for a day or two so I can get the PR done that adds classes, I'd really appreciate it.  It's a massive patch and every rebase is going to be a pain.  I'll post a WIP after I rebase on this patch.

**Comment by rthallisey** [comment]
fixes: https://github.com/openshift/ansible-service-broker/issues/623

**Pull request merged**
